pkg/storage/tree/pprof.go

```diff
@@ -0,0 +1,106 @@
+package tree
+
+import (
+	"io/ioutil"
+
+	"github.com/golang/protobuf/jsonpb"
+	"google.golang.org/protobuf/proto"
+)
+
+type Pprof struct {
+	locations map[string]uint64
+	functions map[string]uint64
+	strings   map[string]int64
+	profile   *Profile
+	tree      *Tree
+	metadata  *PprofMetadata
+}
+
+type PprofMetadata struct {
+	Type      string
+	Unit      string
+	StartTime int64
+	Duration  int64
+}
+
+func (t *Tree) PprofStruct(metadata *PprofMetadata) *Pprof {
+	pprof := &Pprof{
+		locations: make(map[string]uint64),
+		functions: make(map[string]uint64),
+		strings:   make(map[string]int64),
+		profile: &Profile{
+			StringTable: []string{""},
+		},
+		tree:     t,
+		metadata: metadata,
+	}
+	return pprof
+}
+
+func (p *Pprof) Pprof() *Profile {
```

Comment:
Maybe splitting pprof format generation is not needed. An instance of `Pprof` cannot be used to generate the `Profile` more than once (i.e. calling `Pprof()` multiple times would result in different `Profile`s for the same tree), so it may be better to collapse the functionality into a single method with the following signature: `func (t *Tree) Pprof(metadata *PprofMetadata) *Profile`. What do you think?

Reply:
Agree. PprofStruct() is redundant